src/service/acl/mod.rs

```diff
@@ -0,0 +1,107 @@
+                });
+            });
+        self.acl_config.block_list.clone().iter().for_each(|it| {
+            set.insert(AclDatabaseEntry {
+                mode: AclMode::Block,
+                hostname: it.to_owned(),
+            });
+        });
+        match filter {
+            Some(filter) => set.into_iter().filter(|it| it.mode == filter).collect(),
+            None => set.into_iter().collect(),
+        }
+    }
+    pub fn remove_acl(&self, host: Host) -> crate::Result<()> {
+        self.db.remove_acl(host)
+    }
+
+    pub fn add_acl(&self, host: Host, mode: AclMode) -> crate::Result<()> {
+        self.db.add_acl(AclDatabaseEntry {
+            mode: mode,
+            hostname: host,
+        })
+    }
+    /// same as federation_with_allowed however it can work with the fedi_dest type
+    pub fn is_federation_with_allowed_fedi_dest(&self, fedi_dest: &FedDest) -> bool {
+        let hostname = if let Ok(name) = Host::parse(&fedi_dest.hostname()) {
+            name
+        } else {
+            warn!(
+                "cannot deserialise hostname for server with name {:?}",
+                fedi_dest
+            );
+            return false;
+        };
+        return self.is_federation_with_allowed(hostname);
+    }
+
+    /// same as federation_with_allowed however it can work with the fedi_dest type
+    pub fn is_federation_with_allowed_server_name(&self, srv: &ServerName) -> bool {
+        let hostname = if let Ok(name) = Host::parse(srv.host()) {
+            name
+        } else {
+            warn!("cannot deserialise hostname for server with name {:?}", srv);
+            return false;
+        };
+        return self.is_federation_with_allowed(hostname);
+    }
+    /// is federation allowed with this particular server?
+    pub fn is_federation_with_allowed(&self, server_host_name: Host<String>) -> bool {
+        debug!("checking federation allowance for {}", server_host_name);
+        // check blocklist first
+        if self.acl_config.block_list.contains(&server_host_name) {
+            return false;
+        }
+        let mut allow_list_enabled = false;
+        // check allowlist
+        if let Some(list) = &self.acl_config.allow_list {
+            if list.contains(&server_host_name) {
+                return true;
+            }
+            allow_list_enabled = true;
+        }
+
+        //check database
+        match self.db.check_acl(&server_host_name) {
+            Err(error) => {
+                error!("database failed with {}", error);
+                false
+            }
+            Ok(None) if allow_list_enabled => false,
+            Ok(None) => true,
+            Ok(Some(data::AclMode::Block)) => false,
+            Ok(Some(data::AclMode::Allow)) if allow_list_enabled => true,
+            Ok(Some(data::AclMode::Allow)) => {
+                warn!("allowlist value found in database for {} but allow list is not enabled, denied request", server_host_name);
```

Comment:
Am I understanding that if a user configured the allowlist but forgot to enable it, we will be rejecting all other requests?
I don't like that default behaviour, it could potentially be unintended and rejecting all other requests is a drastic thing. The user needs to explicitly enable allowlist/denylist before deciding to reject or allow requests.

Reply:
if a user configured the allowlist in the database but forgot to enable it all the allows turn into blocks I think I can fix that

Reply:
I'd rather just do nothing and log a warning for the admin to see that they need to explicitly opt-in to allowing or denying.

Reply:
okay that is the behavior now, it just logs the warning and lets it through as if it wasn't blocked